common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -739,6 +739,28 @@ bool mon_wlan_hal_dwpal::sta_beacon_11k_request(const SBeaconRequest11k &req, in
         cmd += " ssid=" + req_ssid;
     }
 
+    // use ap_ch_report in the request.
+    if (req.use_optional_ap_ch_report) {
+        std::stringstream ch_report_ss;
+        uint8_t current_channel = 0;
+        for (; current_channel < req.use_optional_ap_ch_report - 1; ++current_channel) {
+            ch_report_ss << +req.ap_ch_report[current_channel] << ",";
+        }
+        ch_report_ss << +req.ap_ch_report[current_channel];
+
+        cmd += " ap_ch_report=" + ch_report_ss.str();
+    } else {
+        LOG(WARNING) << "optinal channel report was not set (might be also ok)";
```

Comment:
> might be also ok

Isn't the definition of optional that it may be missing?

Reply:
I felt uncomfortable with this warning the moment I wrote it and ever since when I see it.

Reply:
You don't issue warnings for missing optional fields. Simply remove it.

Reply:
I'll remove but note that this is a valid situation:

>  you might wanted to set it but something went wrong

In this specific case I can check another field (`channel` I think) and if its value is 255 then it indicates that ap_ch_report _must_ exists.